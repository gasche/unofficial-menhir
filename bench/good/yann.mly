%token LPAREN RPAREN MATRIX TIMES INT PLUS TENSOR
%start<int> exp
%%

exp:
  term {}

fa1: 
  LPAREN exp RPAREN {}
| INT {}

fa4:
  MATRIX {}
| LPAREN exp RPAREN {}

product:
  product TIMES fa1 {}
| fa1 {}

tensorproduct:
  tensorproduct TENSOR fa4 {}
| fa4 {}

term:
  term PLUS product {}
| term PLUS tensorproduct {}
| product {}
| term PLUS tensorproduct {}
