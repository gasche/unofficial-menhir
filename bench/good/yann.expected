%start exp
%token TIMES
%token TENSOR
%token RPAREN
%token PLUS
%token MATRIX
%token LPAREN
%token INT

%type <int> exp
%%

exp:
| term
    {}

fa1:
| LPAREN exp RPAREN
    {}
| INT
    {}

fa4:
| MATRIX
    {}
| LPAREN exp RPAREN
    {}

product:
| product TIMES fa1
    {}
| fa1
    {}

tensorproduct:
| tensorproduct TENSOR fa4
    {}
| fa4
    {}

term:
| term PLUS product
    {}
| term PLUS tensorproduct
    {}
| product
    {}
| term PLUS tensorproduct
    {}

%%



